piq/ssim.py

```diff
@@ -487,30 +412,44 @@ def _ssim(x: torch.Tensor, y: torch.Tensor, kernel: torch.Tensor, data_range: Un
 
     ssim_val = ssim_map.mean(1)
     cs = cs_map.mean(1)
-
-    if full:
-        return ssim_val, cs
-
-    return ssim_val
+    return ssim_val, cs
 
 
 def _multi_scale_ssim(x: torch.Tensor, y: torch.Tensor, data_range: Union[int, float], kernel: torch.Tensor,
                       scale_weights_tensor: torch.Tensor, k1: float, k2: float) -> torch.Tensor:
+    r"""Calculates Multi scale Structural Similarity (MS-SSIM) index for X and Y.
+
+        Args:
+            x: Batch of images, (N,C,H,W).
+            y: Batch of images, (N,C,H,W).
+            data_range: Value range of input images (usually 1.0 or 255).
+            kernel: 2D Gaussian kernel.
+            scale_weights_tensor: Weights for scaled SSIM
+            k1: Algorithm parameter, K1 (small constant, see [1]).
+            k2: Algorithm parameter, K2 (small constant, see [1]).
+                Try a larger K2 constant (e.g. 0.4) if you get a negative or NaN results.
+
+        Returns:
+            Value of Multi scale Structural Similarity (MS-SSIM) index.
+        """
     levels = scale_weights_tensor.size(0)
     min_size = (kernel.size(-1) - 1) * 2 ** (levels - 1) + 1
     if x.size(-1) < min_size or x.size(-2) < min_size:
         raise ValueError(f'Invalid size of the input images, expected at least {min_size}x{min_size}.')
 
     mcs = []
     ssim_val = None
-    for _ in range(levels):
+    for iteration in range(levels):
+        if iteration > 0:
```

Comment:
Why do you need this condition?

Reply:
To compute SSIM on original scale before downsampling it

Reply:
Just changed order of operation to get rid of final downscale, the logic remains the same.